src/bindings/c/include/openvino/c/ov_model.h

```diff
@@ -0,0 +1,148 @@
+ * @param input_node A pointer to the ov_input_node_t.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_input_by_index(const ov_model_t* model, const size_t index, ov_output_const_node_t** input_node);
+
+/**
+ * @brief Returns true if any of the op's defined in the model contains partial shape.
+ * @param model A pointer to the ov_model_t.
+ */
+OPENVINO_C_API(bool) ov_model_is_dynamic(const ov_model_t* model);
+
+/**
+ * @brief Do reshape in model with a list of <name, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_names input tensor name (char *) list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape(const ov_model_t* model,
+                 const char* tensor_names[],
+                 const ov_partial_shape_t* partial_shapes[],
+                 size_t cnt);
+
+/**
+ * @brief Do reshape in model with partial shape for a specified name.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param tensor_name input tensor name (char *).
+ * @param partialShape A PartialShape.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_input_by_name(const ov_model_t* model,
+                               const char* tensor_name,
+                               const ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Do reshape in model for one node(port 0).
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param partialShape A PartialShape.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_one_input(const ov_model_t* model, const ov_partial_shape_t* partial_shape);
+
+/**
+ * @brief Do reshape in model with a list of <port id, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param ports The port list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_by_ports(const ov_model_t* model, size_t* ports, const ov_partial_shape_t** partial_shape, size_t cnt);
+
+/**
+ * @brief Do reshape in model with a list of <ov_output_node_t, partial shape>.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param output_nodes The ov_output_node_t list.
+ * @param partialShape A PartialShape list.
+ * @param cnt The item count in the list.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_model_reshape_by_nodes(const ov_model_t* model,
+                          const ov_output_node_t* output_nodes[],
+                          const ov_partial_shape_t* partial_shapes[],
+                          size_t cnt);
+
+/**
+ * @brief Gets the friendly name for a model.
+ * @ingroup model
+ * @param model A pointer to the ov_model_t.
+ * @param friendly_name the model's friendly name.
+ */
+OPENVINO_C_API(ov_status_e) ov_model_get_friendly_name(const ov_model_t* model, char** friendly_name);
```

Comment:
do we need this in C API>

Reply:
It is required in hello_classification_c, so we have to wrap it to make the sample can run.